src/Microsoft.Health.Dicom.SqlServer/Features/Query/BaseSqlQueryGenerator.cs

```diff
@@ -0,0 +1,229 @@
+                .Append(" AND ")
+                .Append(_parameters.AddParameter(dicomTagSqlEntry.SqlColumn, rangeValueMatchCondition.Maximum.ToString(SqlDateFormat)))
+                .AppendLine();
+        }
+
+        public override void Visit(DateSingleValueMatchCondition dateSingleValueMatchCondition)
+        {
+            var queryTag = dateSingleValueMatchCondition.QueryTag;
+            var dicomTagSqlEntry = DicomTagSqlEntry.GetDicomTagSqlEntry(queryTag);
+            var tableAlias = GetTableAlias(dicomTagSqlEntry, GetKeyFromQueryTag(queryTag));
+            _stringBuilder
+                .Append("AND ");
+
+            AppendExtendedQueryTagKeyFilter(dicomTagSqlEntry, tableAlias, dateSingleValueMatchCondition);
+
+            _stringBuilder
+                .Append(dicomTagSqlEntry.SqlColumn, tableAlias)
+                .Append("=")
+                .Append(_parameters.AddParameter(dicomTagSqlEntry.SqlColumn, dateSingleValueMatchCondition.Value.ToString(SqlDateFormat)))
+                .AppendLine();
+        }
+
+        public override void Visit(PersonNameFuzzyMatchCondition fuzzyMatchCondition)
+        {
+            var queryTag = fuzzyMatchCondition.QueryTag;
+            var dicomTagSqlEntry = DicomTagSqlEntry.GetDicomTagSqlEntry(queryTag);
+            var tableAlias = GetTableAlias(dicomTagSqlEntry, GetKeyFromQueryTag(queryTag));
+
+            var fuzzyMatchString = $"\"{fuzzyMatchCondition.Value}*\"";
+            _stringBuilder
+                .Append("AND ");
+
+            AppendExtendedQueryTagKeyFilter(dicomTagSqlEntry, tableAlias, fuzzyMatchCondition);
+            _stringBuilder
+                .Append("CONTAINS(")
+                .Append(tableAlias)
+                .Append(".")
+                .Append(dicomTagSqlEntry.FullTextIndexColumnName)
+                .Append(", ")
+                .Append(_parameters.AddParameter(dicomTagSqlEntry.SqlColumn, fuzzyMatchString))
+                .Append(")")
+                .AppendLine();
+        }
+
+        private void AppendExtendedQueryTagKeyFilter(DicomTagSqlEntry dicomTagSqlEntry, string tableAlias, QueryFilterCondition filterCondition)
+        {
+            if (dicomTagSqlEntry.IsIndexedQueryTag)
+            {
+                _stringBuilder
+                    .Append(dicomTagSqlEntry.SqlKeyColumn, tableAlias)
+                    .Append("=")
+                    .Append(_parameters.AddParameter(dicomTagSqlEntry.SqlKeyColumn, GetKeyFromQueryTag(filterCondition.QueryTag)))
+                    .AppendLine()
+                    .Append("AND ");
+            }
+        }
+
+        private static int? GetKeyFromQueryTag(QueryTag queryTag)
+        {
+            return queryTag.IsExtendedQueryTag ? queryTag.ExtendedQueryTagStoreEntry.Key : queryTag.IsWorkitemQueryTag ? queryTag.WorkitemQueryTagStoreEntry.Key : null;
```

Comment:
same here.

Reply:
Done made as abstract which is implemented in different generators